mesa/experimental/cell_space.py

```diff
@@ -0,0 +1,390 @@
+import itertools
+import random
+from collections.abc import Iterable
+from functools import cache, cached_property
+from typing import Any, Callable, Optional
+
+from .. import Agent, Model
+
+Coordinate = tuple[int, int]
+
+
+class CellAgent(Agent):
+    """
+    Base class for a model agent in Mesa.
+
+    Attributes:
+        unique_id (int): A unique identifier for this agent.
+        model (Model): A reference to the model instance.
+        self.pos: Position | None = None
+    """
+
+    def __init__(self, unique_id: int, model: Model) -> None:
+        """
+        Create a new agent.
+
+        Args:
+            unique_id (int): A unique identifier for this agent.
+            model (Model): The model instance in which the agent exists.
+        """
+        super().__init__(unique_id, model)
+        self.cell: Cell | None = None
+
+    def move_to(self, cell) -> None:
+        if self.cell is not None:
+            self.cell.remove_agent(self)
+        self.cell = cell
+        cell.add_agent(self)
+
+
+class Cell:
+    __slots__ = [
+        "coordinate",
+        "_connections",
+        "owner",
+        "agents",
+        "capacity",
+        "properties",
+    ]
+
+    def __init__(self, coordinate, owner, capacity: int | None = None) -> None:
+        self.coordinate = coordinate
+        self._connections: list[Cell] = []  # TODO: change to CellCollection?
```

Comment:
I find CellCollection to be different from this. To draw analogy from graphs. A CellCollection is a collection of nodes, and we don't care about their edges structure. But in this case, The `_connections` consists of nodes that are nearest neighbors to this Cell.

If you want to be pedantic, `self._nearest_neighbors`, because `connections` signifies edges instead of nodes.